source/utils/ResourceManager.cpp

```diff
@@ -49,7 +49,7 @@
 #include <OgreRenderTarget.h>
 
 template<> ResourceManager* Ogre::Singleton<ResourceManager>::msSingleton = 0;
-#if defined(OD_DEBUG)
+#if OGRE_PLATFORM == OGRE_PLATFORM_WIN32 && defined(OD_DEBUG)
```

Comment:
Here, why couldn't the application be compiled in debug mode on linux?
I do think this should be the same behaviour for every platform.
Thus, I'd remove the ```OGRE_PLATFORM == OGRE_PLATFORM_WIN32 && ``` part again and would remove the confusing comment as well.

What do you think?